apm-agent-tracer/src/main/java/co/elastic/apm/agent/tracer/reporting/ReportingTracer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.tracer.reporting;
+
+import co.elastic.apm.agent.tracer.Tracer;
+
+import java.io.Closeable;
+import java.util.concurrent.ScheduledExecutorService;
+
+public interface ReportingTracer extends Tracer {
+
+    void addMetric(String name, Labels labels, DoubleSupplier metric);
+
+    void removeMetric(String name, Labels labels);
+
+    /**
+     * Reports an ECS-logging formatted log message.
+     *
+     * @param log log message generated by ecs-logging
+     */
+    void reportLog(String log);
+
+    /**
+     * Reports an ECS-logging formatted log message.
+     *
+     * @param log log message generated by ecs-logging
+     */
+    void reportLog(byte[] log);
+
+    DataWriter newWriter(int maxSerializedSize);
+
+    ScheduledExecutorService getSharedSingleThreadedPool();
```

Comment:
I would move this method to either the general `Tracer` interface or a separate interface.

Reply:
I found a correlation between a plugin reporting and needing a custom lifecycle. Wouldn't it make sense to keep the two together for this reason?

Reply:
Looks like I forgot to add an initial comment, with which my proposal here would make more sense:

I was thinking of splitting `ReportingTracer` into `MetricsTracer` and `LogReportingTracer`. Because both would still need the `getSharedSingleThreadedPool()` (I think?) I would move that to the parent interface.